### PR TITLE
[FEATURE] Allow optional expectation_suite_name to be passed to RunDataAssistantEvent

### PR DESCRIPTION
**Pull request by TrangPham**
This PR allows expectation_suite_name to be passed in optionally when running DA from RunDataAssistantEvent

We need to ensure that the current events without expectation_suite_name are still able to be processed as FE is already sending those events for RunOnboardingDataAssistant

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [x] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: great_expectations/agent/actions/data_assistants/utils.py [hunk 1]
```diff
@@ -47,7 +47,11 @@ def build_action_result(
     data_assistant_result: DataAssistantResult,
     id: str,
 ) -> ActionResult:
-    expectation_suite_name = f"{event.type} {event.data_asset_name} assistant suite"
+    if event.expectation_suite_name is not None:
+        expectation_suite_name = event.expectation_suite_name
+    else:
+        expectation_suite_name = f"{event.type} {event.data_asset_name} assistant suite"
```
Comment (joshua-stauffer) [inline great_expectations/agent/actions/data_assistants/utils.py]: do we want the event type to be in this name? That seems like a implementation detail that doesn't help the user much - i'm imagining this creating a suite named something like `missingness_data_assistant_request.received orders assistant suite`.

**Review by joshua-stauffer — CHANGES_REQUESTED** [review]
one comment - happy to re-review later today!

File: great_expectations/agent/actions/data_assistants/utils.py (hunk 1 above)
Comment (TrangPham) [reply great_expectations/agent/actions/data_assistants/utils.py]: Hmmm good point I'll update

**Review by JennyTee — APPROVED** [review]

**Review by Shinnnyshinshin — APPROVED** [review]

**Review by joshua-stauffer — APPROVED** [review]
thanks for addressing, looks great!

**Pull request merged**